ArkKit/ark-physics-kit/ArkPhysicsSystem.swift

```diff
@@ -0,0 +1,232 @@
+import Foundation
+
+/**
+ *  A system which updates all running animation instances in ArkECS after the given delta.
+ */
+class ArkPhysicsSystem: System {
+    var active: Bool
+    var scene: AbstractArkGameScene
+    var eventManager: ArkEventManager
+
+    init(active: Bool = true, gameScene: AbstractArkGameScene, eventManager: ArkEventManager) {
+        self.active = active
+        self.scene = gameScene
+        self.eventManager = eventManager
+    }
+    
+    func update(deltaTime: TimeInterval, arkECS: ArkECS) {
+        let physicsComponents = getPhysicsComponents(arkECS)
+        syncToPhysicsEngine(physicsComponents, arkECS: arkECS)
+        scene.update(deltaTime)
+        syncFromPhysicsEngine(arkECS: arkECS)
+    }
+    
+    private func setupPhysicsScene(){
+        scene.sceneUpdateDelegate = self
+    }
+    
+    private func getPhysicsComponents(_ arkECS: ArkECS) -> [(Entity, PhysicsComponent)] {
+        return arkECS.getEntities(with: [PhysicsComponent.self]).compactMap { entity in
+            guard let physicsComponent = arkECS.getComponent(ofType: PhysicsComponent.self, for: entity) else {
+                return nil
+            }
+            return (entity, physicsComponent)
+        }
+    }
+    
+    func syncFromPhysicsEngine(arkECS: ArkECS) {
+        let syncStrategies: [ComponentSyncing] = [PhysicsComponentSync(), PositionComponentSync(), RotationComponentSync()]
+        
+        scene.forEachEntity(perform: { entityId, physicsBody in
+            syncStrategies.forEach { strategy in
+                strategy.sync(entityId: entityId, with: physicsBody, using: arkECS)
+            }
+        })
+    }
+    
+    func syncToPhysicsEngine(_ physicsComponents: [(Entity, PhysicsComponent)], arkECS: ArkECS) {
+        for (entity, physics) in physicsComponents {
+            handlePhysicsComponentRemovalIfNeeded(for: entity, using: physics, arkECS: arkECS)
+            
+            if physics.toBeRemoved {
+                continue
+            }
```

Comment:
The logic flow together with the code for `handlePhysicsComponentRemovalIfNeeded` feels odd to me, like if it's to be removed you don't need to handle the removal anymore? Can ignore if future changes requires this ordering of checking

Reply:
the toBeRemoved mark is done by the the dev side, this one TBD, just leaving it in the PhysicsSystem as it can handle the purging, unless we have a RemoveSystem somewhere @didymental I'm assuming that if the Entity is removed, the Renderable component is in turn removed, purging it from the view as well